src/errors.ts

```diff
@@ -45,6 +45,17 @@ export function isResponseError(error: unknown): error is AnyResponseError {
   );
 }
 
+export async function extractResponseBody(error: unknown): Promise<any> {
+  if (isResponseError(error)) {
+    const responseError = error as AnyResponseError;
+    const respJson = await responseError.response.clone().json();
```

Comment:
May need a try/catch around here in the event we get some non-JSON response with a fallback to `await responseError.response.clone().text()`